src/record_syscall.cc

```diff
@@ -3749,6 +3749,20 @@ static Switchable rec_prepare_syscall_arch(RecordTask* t,
       return ALLOW_SWITCH;
     }
 
+    case Arch::modify_ldt: {
+      int func = regs.arg1();
+      if (func == 0 || func == 2) {
+        syscall_state.reg_parameter(
+            2, ParamSize::from_syscall_result<typename Arch::ssize_t>(
```

Comment:
Upon reading the man page again, I also realized that unlike set_thread_area, modify_ldt does not poke `entry_number`, so I can the else branch here.